### PR TITLE
Update Summit build doc and machine file

### PR DESCRIPTION
**Pull request by pgrete**
## PR Summary

Updates Summit doc and machine file.
Does include #435 so should be merged afterwards.

## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented.
- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md

### CODE REVIEW

File: cmake/machinecfg/Summit.cmake [hunk 1]
```diff
@@ -18,20 +18,36 @@
...
+set(TEST_NUMPROC_FLAG "-a" CACHE STRING "Flag to set number of processes")
+set(NUM_GPU_DEVICES_PER_NODE "6" CACHE STRING "6x V100 per node")
+set(PARTHENON_ENABLE_GPU_MPI_CHECKS OFF CACHE BOOL "Disable check by default")
+
 if (${MACHINE_VARIANT} MATCHES "mpi")
-  set(TEST_MPIEXEC jsrun CACHE STRING "Command to launch MPI applications")
-  set(TEST_NUMPROC_FLAG "-a" CACHE STRING "Flag to set number of processes")
+  # Use a single resource set on a node that includes all cores and GPUs.
+  # GPUs are automatically assigned round robin when run with more than one rank.
   list(APPEND TEST_MPIOPTS "-n" "1" "-g" "6" "-c" "42" "-r" "1" "-d" "packed" "-b" "packed:7" "--smpiargs='-gpu'")
```
Comment (JoshuaSBrown) [inline cmake/machinecfg/Summit.cmake:50]: So if you are just running with mpi I'm guessing the gpu's just won't be called even though they are passed to mpi?

File: tst/regression/run_test.py [hunk 2]
```diff
@@ -138,6 +138,11 @@ class TestError(RuntimeError):
                         nargs=1,
                         help='change MPI run wrapper command (e.g. for job schedulers)')
 
+    parser.add_argument('--mpirun_ranks',
+                        default=[],
+                        action='append',
+                        help='add option to specify number of ranks')
+
```
Comment (JoshuaSBrown) [inline tst/regression/run_test.py]: Why not add

```
default=1,
nargs=1,
```

Won't there always be a number passed in with the flag?

File: cmake/machinecfg/Summit.cmake (hunk 1 above)
Comment (pgrete) [reply cmake/machinecfg/Summit.cmake:50]: What you do you refer to with "GPUs being called"?
These option ensure that there's a resource set for a node that includes all resources (all CPU cores and all GPUs).
The number of ranks that are launched in the tests is controlled via the `-a` flag and the number of GPUs controlled via the `kokkos-num-devices` argument.

File: tst/regression/utils/test_case.py [hunk 3]
```diff
@@ -106,20 +106,10 @@ def __init__(self,run_test_path,**kwargs):
...
-        for s in argstrings:
-          if s in self.parameters.mpi_opts:
-            index = self.parameters.mpi_opts.index(s)
-            if index < len(self.parameters.mpi_opts) - 1:
-              try:
-                self.parameters.num_ranks = int(self.parameters.mpi_opts[index+1])
-              except ValueError:
-                pass
+        if len(self.parameters.mpi_ranks) == 2:
+            self.parameters.num_ranks = int(self.parameters.mpi_ranks[1])
```
Comment (JoshuaSBrown) [inline tst/regression/utils/test_case.py]: When will num_ranks be different from mpi_ranks? If the array is empty I see it but what about if mpi_ranks == 4?

File: tst/regression/run_test.py (hunk 2 above)
Comment (pgrete) [reply tst/regression/run_test.py]: While in principle I'd be great to enforce this, I'm not sure we can given our logic in the following.
For example, for mpirun the effective line for `run_test.py` reads
`run_test.py --mpirun=mpirun --mpirun_ranks=-n --mpirun_ranks=4 ...`
whereas for jsrun is reads like
`run_test.py --mpirun=jsrun --mpirun_ranks=-a --mpirun_ranks=4 ...`

One option would be to split mpirun_ranks even further and separate the flag (`-n`, `-a`, ...) from the the actual number.

File: tst/regression/utils/test_case.py (hunk 3 above)
Comment (pgrete) [reply tst/regression/utils/test_case.py]: As mentioned in the comment above mpirun_ranks will always contain two entries. One for the flag and one for the number.

File: tst/regression/run_test.py (hunk 2 above)
Comment (Yurlungur) [reply tst/regression/run_test.py]: I would be in favor of splitting `mpirun_ranks` into `mpirun_rank_flag` and `mpirun_num_ranks` or something.

**Review by Yurlungur — APPROVED** [review]
Overall looks good, assuming you tested it. My comment below should be considered non-blocking.

File: tst/regression/run_test.py (hunk 2 above)
Comment (pgrete) [reply tst/regression/run_test.py]: Done now #435

**Comment by pgrete** [comment]
> Overall looks good, assuming you tested it. My comment below should be considered non-blocking.

Yes, I tested on Summit and the tests work as described in the build doc and also run on multiple GPUs.

File: tst/regression/run_test.py (hunk 2 above)
Comment (JoshuaSBrown) [reply tst/regression/run_test.py]: Great.

File: tst/regression/utils/test_case.py (hunk 3 above)
Comment (JoshuaSBrown) [reply tst/regression/utils/test_case.py]: Oh, I completely parsed over len, disregard my comment.

File: cmake/machinecfg/Summit.cmake (hunk 1 above)
Comment (JoshuaSBrown) [reply cmake/machinecfg/Summit.cmake:50]: I'm saying why do you need "-g 6" if you are just running with the "mpi" variant, but I guess it doesn't really matter. If Kokkos is disabled anyway the gpus won't be utilized. Regardless, it's a non-issue.

**Review by JoshuaSBrown — APPROVED** [review]

**Pull request merged**